src/controller/auth.js

```diff
@@ -1,32 +1,80 @@
 import express from 'express';
+import dotenv from 'dotenv';
 import cors from 'cors';
 import bodyParser from 'body-parser';
+import * as admin from 'firebase-admin';
+import path from 'path';
+import { createNewUser, getAllUsers } from '../service/service';
 
 const AuthController = express.Router();
 AuthController.use(cors({ origin: '*' }));
 AuthController.use(bodyParser.json());
+dotenv.config({ path: path.join(__dirname, '..', '.env') });
 
 /**
  * Sign-in endpoint
  */
-AuthController.post('/api/v1/auth/sign-in', (req, res) => {
-  const username = req?.body?.username;
+AuthController.post('/api/v1/auth/sign-in', async (req, res) => {
+  // get firebase auth variables
+  const email = req?.body?.email;
   const password = req?.body?.password;
 
-  if (!username) {
-    res.status(400).json({ message: 'no username provided in request to server' });
-    return;
-  }
-
-  if (!password) {
+  if (!email) {
+    res.status(400).json({ message: 'no email provided in request to server' });
+  } else if (!password) {
     res.status(400).json({ message: 'no password provided in request to server' });
-    return;
+  } else {
+    console.log('success');
   }
+});
 
-  // TODO: implement sign-in endpoint
-  console.log('credentials received: ', username, password);
-
-  res.status(400).json({ message: 'invalid username or password' });
+/**
+ * Sign-up endpoint
+ */
+AuthController.post('/api/v1/auth/sign-up', async (req, res) => {
+  const user = req?.body?.username;
+  const ema = req?.body?.email;
+  const pass = req?.body?.password;
+  const cpass = req?.body?.cpassword;
+  const avalUser = await getAllUsers(user);
+  if (ema) {
```

Comment:
Hello, it's possible to clean up this code by doing an early return.

So like:

```
if (!user) {
res.status(400).json();
return;
}

if (!pass) {
res.status(400).json();
return;
}

...

// then success code goes here
res.status(200).json(...);
```

Reply:
Yeah, I will clean it up.